Add support with ss3 version of SimpliSafe REST APIs

chowielin opened the pull request:
No NPM package published yet. In install.sh, add:
npm install -g git+https://github.com/chowielin/homebridge-simplisafe-security-system.git

chowielin committed:
Support ss3 version of SimpliSafe REST APIs

chowielin committed:
Add missing NPM deps

enriquecaballero commented:
+1

chowielin committed:
Do auth check to test if token is still good, else reacquire it. Add some logging on errors

chowielin committed:
Fix infinite loop

chowielin committed:
Add retry for get alarm state

chowielin committed:
Add retries count

chowielin committed:
Set token in headers after reinit of token

chowielin committed:
Make it so that only one token reacquire ever runs at a time. Other reacquire attempts will wait and see the new token is fresh and not attempt to reacquire again.

eflee committed:
Fixing client to filter our inactive subscriptions (which have invalid states).

chowielin committed:
Merge pull request #1 from eflee/master

Fixing client to filter our inactive subscriptions

enriquecaballero commented:
Just thought I'd throw this out there -- this project hasn't been updated in about 2 years and I have no idea if it's maintained anymore; I wouldn't wait on this getting merged. If no response happens, just release it yourself under another name.

chowielin commented:
@enriquecaballero ok yeah I noticed. I will probably go ahead and publish a separate NPM then. I will also look to separate the SS3 client to a separate library so others can use it outside of Homebridge.

enriquecaballero commented:
@chowielin Cool. I started looking into maybe getting the separate (motion) sensors to show up in HomeKit as well. Looks like there's an endpoint that returns JSON with all the sensors tied to the system.

chowielin commented:
@enriquecaballero cool. I noticed that as well and added that function in the SS3Client called "getSensors". I saw it gives status for doors being opened or not.

chowielin committed:
Remove internal ss3-client and specify simplisafe-ss3 as dependency

chowielin committed:
Update ss client dependency version

chowielin committed:
Update README with correct commandline for installation

chowielin committed:
Map ALARM_COUNT to SecuritySystemTargetState.AWAY_ARM

To handle case where alarm is triggered and the status was not being reported correctly. ALARM_COUNT doesn't denote whether home mode or away mode was triggered so just defaulting to away mode.

mbriney commented:
It seems the update SimpliSafe did the other day has removed access to the previous APIs.  It gets a token but doesn't update status properly.  I'd recommend merging in the code from @chowielin as pulling his repository fixes the package.

ddoria921 commented:
@chowielin are you still thinking about publishing this as a separate npm package?

chowielin commented:
@ddoria921 It’s on my radar. I forgot about doing that! Glad people are finding it useful.

ddoria921 commented:
Ping me when it’s published. I want to contribute and I figured I might as well pitch in on the more recently miantened package. Thanks for keeping it up-to-date!

fholgado commented:
D'oh. I don't know why GitHub isn't sending me emails about this. Sorry y'all, would have totally merged this in sooner :(

chowielin committed:
Update README.md
